projects/optic/src/__tests__/integration/ruleset-publish.test.ts

```diff
@@ -0,0 +1,31 @@
+import { runOptic, setupWorkspace, normalizeWorkspace } from './integration';
+
+jest.setTimeout(30000);
+// TODO replace this with real mocks when connected to backend
+jest.mock('node-fetch');
+
+describe('optic ruleset publish', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  test('can publish a ruleset', async () => {
+    const workspace = await setupWorkspace('ruleset-publish/valid-js-file');
+    const { combined, code } = await runOptic(
+      workspace,
+      'ruleset publish ./rules.js --token atoken'
```

Comment:
I think it'd be better to lean on an env var than a token flag on the CLI, just so someone can opt into having their token in cleartext on their command line.